convert2rhel/toolopts.py

```diff
@@ -580,5 +656,29 @@ def _validate_serverurl_parsing(url_parts):
     return url_parts
 
 
+def _organize_cli_options(argv):
+    """Organize the command line options used with the tool"""
+    # if an arg is a command or parent parser (_parser) we need to reorganize them to be infront of the other args
+    args = []
+    arg_pos = 0
+    if len(argv):
+        # Add each argument based on the position of the parent options
+        # Reorder each to be before options other than parent ones.
+        for index, argument in enumerate(argv):
+            if argument in PARENT_ARGS and argv[index - 1] not in ARGS_WITH_VALUES:
+                args.insert(arg_pos, argument)
+                arg_pos += 1
+            # If the argument is either `convert` or `analyze` and not after
+            # an argument that accepts a value, then we do the reordering
+            elif argument in ("convert", "analyze") and argv[index - 1] not in ARGS_WITH_VALUES:
+                args.insert(arg_pos, argument)
+            else:
+                # Append here anything else that is in the argv.
+                args.append(argument)
+    else:
+        args.append("convert")
+    return args
```

Comment:
```suggestion
def _add_default_command(argv):
    """Add the default command when none is given"""
needs_convert = True
for index, argument in enumerate(argv):
    if argument in ("convert", "analyze"):
        need_to_add_convert = False
        return
    if not argument in PARENT_ARGS and argv[index - 1] in ARGS_WITH_VALUES:
        need_convert = True
        break

if need_convert:
    argv.insert(0, "convert")
```

I think the above does what we need to make "convert" the default command when no command is given.  

There is one case that will raise an error in this code that the `organize cli_options` function will make into a valid command line instead; if the user specifies a command but they specify it in the wrong place.  For instance, at the end of the list of commands:
```
convert2rhel --no-rhsm --no-rpm-va convert
# Will be reorganized to convert2rhel convert --no-rhsm --no-rpm-va
```

Since this is a user error I think argparse yielding an error in that is perfectly acceptable.

Rodolfo, I know you were a big proponent of rewriting the order of the arguments instead of simply determining whether to add convert as the first element or not.  Do you think that simplicity wins out here or correcting the user's placement of the command?

Reply:
@r0x0d ^

Reply:
After we discussed that in a meeting, I think I'm leaning towards simplicity and let argparse tell the user they did something wrong. 

I like the idea of organizing the user CLI to make sure that in any way they can run convert2rhel it will work the correct way, but then, it will allow people to do whatever they want on CLI leave, like, placing `convert` at the beginning, or in the middle or the end of the command, which, sounds terrible for parsing. 

I think we should go with this simple solution and let argparse say to the user that they are using it incorrectly.

Also, the code looks very readable as well.

Reply:
I will go ahead with the simplified function then, thanks everyone